python/foglamp/services/south/server.py

```diff
@@ -232,7 +235,7 @@ def run(self):
             for task_pending in pending:
                 task_pending.cancel()
             await asyncio.sleep(2)
-        except (asyncio.CancelledError, exceptions.DataRetrievalError):
+        except asyncio.CancelledError:
```

Comment:
move this after `task_pending.cancel()`  inside for loop 

```
for task in pending:
        task.cancel()
        # Now, await task to execute it's cancellation.
        # Cancelled task raises asyncio.CancelledError that we can suppress:
        with suppress(asyncio.CancelledError):
            await task_pending
```


 make  await asyncio.sleep(2) to bit low. may be 1. or 0.5 and inside for loop

Reply:
Using "with suppress(asyncio.CancelledError)" needs additional dependency contextlib.suppress. Also it provides absolutely no difference in functionality, so, IMHO, we are better off with current version only.

Reply:
let's remove suppress!

Reply:
@praveen-garg note that asyncio.CancelledError is raised after the task is cancelled. Hence the correct code is as below:
```
for task_pending in pending:
        try:
             task_pending.cancel()
             # Now, await task  to execute it's cancellation.
             # Cancelled task raises asyncio.CancelledError:
        except asyncio.CancelledError:
             pass
```